tensorboard/webapp/metrics/views/card_renderer/scalar_card_component.ts

```diff
@@ -185,7 +193,51 @@ export class ScalarCardComponent {
     }
   }
 
-  trackByTooltipDatum(index: number, datum: TooltipDatum) {
+  trackByTooltipDatum(index: number, datum: ScalarTooltipDatum) {
     return datum.id;
   }
+
+  getCursorAwareTooltipData(
+    tooltipData: TooltipDatum<ScalarCardSeriesMetadata>[],
+    cursorLoc: {x: number; y: number}
+  ): ScalarTooltipDatum[] {
+    const scalarTooltipData = tooltipData.map((datum) => {
+      return {
+        ...datum,
+        metadata: {
+          ...datum.metadata,
+          closest: false,
+          distSqToCursor:
+            (datum.point.x - cursorLoc.x) ** 2 +
+            (datum.point.y - cursorLoc.y) ** 2,
```

Comment:
Optional: could use `Math.hypot` here, which you might find simpler? On
the perf side, you have an unneeded `fsqrt`, but in exchange all the
arithmetic is in native code. May or may not matter either way.

Reply:
Thanks, I actually forgot about existence of hypot :) Applied.